### DISCUSSION TITLE
CAT: Enhance Value selection for EntityReference parameter type

### DISCUSSION THREAD

**Discussion by CasimirForelli** [rappen/CustomActionTester#30] [Ideas]
Hi @rappen ,

I have some suggestions for enhancing the value selection for parameters of type EntityReference for the Custom Action Tester.
Currently you can only use the "Lookup" dialog and use System Views to pick the needed record.
This is nice and convenient is many cases, but if you have to pick a record from a very large entity you are lost, because you cannot filter properly or it takes very long until the view loads. And even then filtering does not work afterwards.
So my suggestion would be to simple allow the entering and usage of the record GUID as alternative, like it is possible in the FetchXml Builder.
Alternatively, or in addition, it would also help if the list of Views in the Lookup dialog would also contain the personal views of the current user. This will give us he chance to pre-filter the records needed for a test and store it as view.

Thank you and kind regards
Casimir

PS: really appreciating all your work for XrmToolbox, Thanks a lot

**Comment by rappen** [comment]
Great ideas, thanks!
Creating an issue for this, we'll see when I find time to implement it.

**Reply by rappen** [reply]
Now in issue #31
